components/patient-measurements/api/src/test/java/org/phenotips/measurements/internal/MeasurementAgeUpdaterTest.java

```diff
@@ -0,0 +1,186 @@
+
+        verify(this.measurement).setFloatValue(AGE_PROPERTY_NAME, 0);
+        verify(this.measurement).removeField(DATE_PROPERTY_NAME);
+    }
+
+    @Test
+    public void checkAgeIsCorrectlyCalculated() throws ParseException, ComponentLookupException
+    {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
+
+        String birthDateString = "01-01-2015 12:00:00";
+        Date birthDate = dateFormat.parse(birthDateString);
+        when(this.patientRecordObj.getDateValue(DATE_OF_BIRTH_PROPERTY_NAME)).thenReturn(birthDate);
+
+        String measureDateString = "01-01-2016 12:00:00";
+        Date measureDate = dateFormat.parse(measureDateString);
+
+        when(this.measurement.getDateValue(DATE_PROPERTY_NAME)).thenReturn(measureDate);
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+        verify(this.measurement).setFloatValue(eq(AGE_PROPERTY_NAME), eq(365/30.4375f));
+
+        measureDateString = "01-01-2035 12:00:00";
+        measureDate = dateFormat.parse(measureDateString);
+
+        when(this.measurement.getDateValue(DATE_PROPERTY_NAME)).thenReturn(measureDate);
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+        verify(this.measurement).setFloatValue(eq(AGE_PROPERTY_NAME), eq((365*20 + 5)/30.4375f));
+
+        measureDateString = "01-03-2016 12:00:00";
+        measureDate = dateFormat.parse(measureDateString);
+
+        when(this.measurement.getDateValue(DATE_PROPERTY_NAME)).thenReturn(measureDate);
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+        verify(this.measurement).setFloatValue(eq(AGE_PROPERTY_NAME), eq((31 + 29 + 365)/30.4375f));
+
+    }
+
+    @Test
+    public void checkMultipleDatesAreUpdated() throws ComponentLookupException {
+        BaseObject measurement2 = mock(BaseObject.class);
+        BaseObject measurement3 = mock(BaseObject.class);
+        this.objects.add(measurement2);
+        this.objects.add(measurement3);
+
+        when(measurement2.getDateValue(DATE_PROPERTY_NAME)).thenReturn(new Date());
+        when(measurement3.getDateValue(DATE_PROPERTY_NAME)).thenReturn(new Date());
+
+        this.mocker.getComponentUnderTest().onEvent(event, source, data);
+        verify(this.measurement).setFloatValue(eq(AGE_PROPERTY_NAME), anyFloat());
+        verify(measurement2).setFloatValue(eq(AGE_PROPERTY_NAME), anyFloat());
+        verify(measurement3).setFloatValue(eq(AGE_PROPERTY_NAME), anyFloat());
+    }
+
+    @Test
+    public void returnsNormallyWhenGivenNoMeasurementObjects() throws Exception {
+        when(this.source.getXObjects(any(EntityReference.class))).thenReturn(null);
+        NullPointerException exceptionWhenListIsNull = null;
+        try {
+            this.mocker.getComponentUnderTest().onEvent(event, source, data);
+        } catch(NullPointerException e){
+            exceptionWhenListIsNull = e;
```

Comment:
You don't need to do this. If there is an exception thrown, the test will automatically fail.

If, however, you do want to test that calling a method will throw an exception, then you would annotate the method with:

    @Test(expected=NullPointerException.class)